pkg/alertmanager/alertmanager.go

```diff
@@ -342,6 +345,33 @@ func (am *Alertmanager) GetReceiversHandler(w http.ResponseWriter, _ *http.Reque
 	}
 }
 
+func (am *Alertmanager) TestTemplatesHandler(w http.ResponseWriter, r *http.Request) {
+	c := alertingNotify.TestTemplatesConfigBodyParams{}
+	if err := json.NewDecoder(r.Body).Decode(&c); err != nil {
+		http.Error(w,
+			fmt.Sprintf("error unmarshalling test templates config JSON: %s", err.Error()),
+			http.StatusBadRequest)
+	}
+
+	am.templatesMtx.RLock()
+	tmpls := make([]alertingTemplates.TemplateDefinition, len(am.templates))
+	copy(tmpls, am.templates)
+	am.templatesMtx.RUnlock()
+
+	response, err := alertingNotify.TestTemplate(r.Context(), c, tmpls, am.cfg.ExternalURL.String(), am.logger)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("error testing templates: %s", err.Error()),
+			http.StatusInternalServerError)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
```

Comment:
We're not returning after calling `http.Error()`, so the function continues executing and writing to the response writer. HTTP response I got (summarized):

```bash
HTTP/1.1 400 Bad Request
Content-Type: application/json

error unmarshalling test templates config JSON: invalid character '}' after object key
{"Results":null,"Errors":null}
```

In any case, submitting a bad request to Grafana's current template testing endpoint results in a JSON response. Any reason why we're not following the same pattern here?